modules/effects/src/concat_latest_from.ts

```diff
@@ -0,0 +1,38 @@
+import { Observable, of, OperatorFunction } from 'rxjs';
+import { concatMap, withLatestFrom } from 'rxjs/operators';
+
+type TypeOfObservable<T> = T extends Observable<infer U> ? U : never;
+
+/**
+ * 'concatLatestFrom' combines the source value
+ * and the last available value from a lazily evaluated Observable
+ * in a new array
+ */
+export function concatLatestFrom<T extends Observable<unknown>[], V>(
```

Comment:
I'd add the comment here, explaining why overload `extends Observable<unknown>[]` is used first.

E.g. It took me some time to figure that out. I was combining everything into just a single function, but that was breaking the proper arguments types order.

```suggestion
// The array overload is needed first because we want to maintain the proper order in the resulting tuple
export function concatLatestFrom<T extends Observable<unknown>[], V>(
```